node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -737,6 +859,331 @@ mod tests {
+            ("MASQ_REAL_USER", "999:999:/home/malooga"),
+        ]
+        .into_iter()
+        .for_each(|(name, value)| std::env::set_var(name, value));
+        let args = ArgsBuilder::new()
+            .param("--real-user", "1001:1001:cooga")
+            .param("--data-directory", &home_dir.to_string_lossy().to_string());
+        let args_vec: Vec<String> = args.into();
+        let dir_wrapper = DirsWrapperMock::new()
+            .home_dir_result(Some(home_dir.clone()))
+            .data_dir_result(Some(data_dir.to_path_buf()));
+
+        let result = server_initializer_collected_params(&dir_wrapper, args_vec.as_slice());
+        let multiconfig = result.unwrap();
+
+        assert_eq!(multiconfig.is_user_specified("--data-directory"), true);
+        assert_eq!(
+            &value_m!(multiconfig, "data-directory", String).unwrap(),
+            &home_dir.to_string_lossy().to_string()
+        );
+        #[cfg(not(target_os = "windows"))]
+        assert_eq!(
+            &value_m!(multiconfig, "real-user", String).unwrap(),
+            "1001:1001:cooga"
+        );
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "You need to define data-directory to define config file with naked directory 'dirname/config.toml'."
+    )]
+    fn server_initializer_collected_params_fails_on_naked_dir_config_file_without_data_directory() {
+        running_test();
+        let _guard = EnvironmentGuard::new();
+        let _clap_guard = ClapGuard::new();
+        let home_dir = ensure_node_home_directory_exists( "node_configurator_standard","server_initializer_collected_params_handle_config_file_from_environment_and_real_user_from_config_file_with_path_started_by_dot");
+
+        let data_dir = &home_dir.join("data_dir");
+        vec![("MASQ_CONFIG_FILE", "config/config.toml")]
+            .into_iter()
+            .for_each(|(name, value)| std::env::set_var(name, value));
+        let args = ArgsBuilder::new();
+        let args_vec: Vec<String> = args.into();
+        let dir_wrapper = DirsWrapperMock::new()
+            .home_dir_result(Some(home_dir.clone()))
+            .data_dir_result(Some(data_dir.to_path_buf()));
+
+        let _result = server_initializer_collected_params(&dir_wrapper, args_vec.as_slice());
+    }
+
+    #[test]
+    fn server_initializer_collected_params_combine_vlcs_properly() {
+        running_test();
+        let _guard = EnvironmentGuard::new();
+        let _clap_guard = ClapGuard::new();
+        let home_dir = ensure_node_home_directory_exists(
+            "node_configurator_standard",
+            "server_initializer_collected_params_combine_vlcs_properly",
+        );
+        let data_dir = &home_dir.join("data_dir");
+        let config_file = File::create(&home_dir.join("config.toml")).unwrap();
+        let current_directory = current_dir().unwrap();
+        fill_up_config_file(config_file);
+
+        let env_vec_array = vec![
+            ("MASQ_CONFIG_FILE", "config.toml"),
```

Comment:
`config.toml` is a default. Can you specify something different here to prove that the default isn't leaking through somehow?